### PR TITLE
[Bugfix] Top offset is incorrect when a user sends a couple of messages and edits second message

### PR DESCRIPTION
**Pull request by emlynmac**
# What
When editing messages, top margin gets expanded when the message is in the middle of a group.
image

Updated some styles to get:
Screenshot 2024-02-07 at 3 30 07 PM

# How Tested
Storybook, CI

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/styles/ChatMessageComponent.styles.ts [hunk 1]
```diff
@@ -125,9 +125,7 @@ export const chatMessageMenuStyle = mergeStyles({
  * @private
  */
 export const useChatMessageEditContainerStyles = makeStyles({
-  root: {
-    paddingTop: '1.25rem' //height of the menu button + marginBottom
-  },
+  root: {},
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/styles/ChatMessageComponent.styles.ts:128]: We can delete this `root` styles as it's empty

File: packages/react-components/src/components/styles/ChatMessageComponent.styles.ts [hunk 2]
```diff
@@ -136,16 +134,19 @@ export const useChatMessageEditContainerStyles = makeStyles({
     ...shorthands.borderRadius(tokens.borderRadiusMedium),
     // The border thickness changes on hover, to prevent the border thickness change causing the
     // input box to shift we apply a margin to compensate. This margin is then removed on hover when the border is thicker.
-    ...shorthands.margin(`${defaultSendBoxActiveBorderThicknessREM - defaultSendBoxInactiveBorderThicknessREM}rem`),
+    marginBottom: '2px',
+    marginTop: '1rem',
     // Width should be updated on hover to include the border width change
     width: `calc(100% - ${defaultSendBoxActiveBorderThicknessREM}rem)`,
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/styles/ChatMessageComponent.styles.ts:140]: This change breaks the current UI, the border makes the edit box to change position for a pixel or so on hover event. Steps to reproduce:
- open MessageThread preview in Storybook
- click on edit button for "your" message
- change focus from the message
- hover over the edit box (the blue border will shown) 
- hover over other messages
- check how it works on the current main branch

**Review by vhuseinova-msft — APPROVED** [review]

**Review by Leah-Xia-Microsoft — APPROVED** [review]
Looks good

File: packages/react-components/src/components/styles/ChatMessageComponent.styles.ts [hunk 3]
```diff
@@ -136,16 +133,25 @@ export const useChatMessageEditContainerStyles = makeStyles({
...
     '&:hover, &:active, &:focus, &:focus-within': {
-      ...shorthands.margin('0rem'),
       ...shorthands.borderWidth(`${defaultSendBoxActiveBorderThicknessREM}rem`),
+      ...shorthands.margin('16px', 0, 0, 0),
       width: '100%'
     }
   },
+  bodyAttached: {
+    marginBottom: '1px',
+    marginTop: '3px',
```
[the remark this answers is not in the document]
Comment (emlynmac) [reply packages/react-components/src/components/styles/ChatMessageComponent.styles.ts:147]: I've updated the spacing to the original size.

**Pull request merged**